library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -260,4 +260,15 @@ void persist() {
                 downloadBatchStatus.notificationSeen()
         );
     }
+
+    void persist() {
```

Comment:
All this switching between `sync` and `async` is hurting my head. Would it be better to have an executor that lives at the db level so that we can queue these things? Rather than switching on the higher levels?

Reply:
Wait, the executor does live on the persistence 😕